parser.go

```diff
@@ -1488,27 +1486,70 @@ func (p *parser) parseSchemaPropertiesFromStructFields(pkgPath, pkgName string,
 		isSliceOrMap := strings.HasPrefix(typeAsString, "[]") || strings.HasPrefix(typeAsString, "map[]")
 		isInterface := strings.HasPrefix(typeAsString, "interface{}")
 		if isSliceOrMap || isInterface || typeAsString == "time.Time" {
-			var err error
-			fieldSchema, err = p.parseSchemaObject(pkgPath, pkgName, typeAsString, true)
-			if err != nil {
-				p.debug(err)
-				return
+			splitType := strings.Split(typeAsString, "]")
+			if len(splitType) > 1 {
+				valueType := splitType[1]
+				if !isBasicGoType(valueType) {
+					if _, ok := p.KnownIDSchema[valueType]; ok {
+						if strings.HasPrefix(splitType[0], "map") {
```

Comment:
Branching logic nested 5 layers deep inside a for loop? My brain is leaking out of my ears XD

Any chance we can break this logic out into more than one function, maybe separate the concerns a little?

Reply:
It's OK, I walked through it line by line and I think I get it now 👍  We can punt on breaking it up into smaller functions till we have the bandwidth for a larger effort to that effect later on down the line.